src/parent.ts

```diff
@@ -0,0 +1,82 @@
+import Undici from "undici";
+import { RequestOptions } from "undici/types/dispatcher";
+import {
+  ActivityResponse,
+  AnnouncementsResponse,
+  BadgesResponse,
+  BehaviourResponse,
+  DetentionsResponse,
+  GetActivityOptions,
+  GetBehaviourOptions,
+  GetHomeworkOptions,
+  GetLessonsOptions,
+  Homework,
+  HomeworksResponse,
+  LessonsResponse,
+  Student,
+} from "./types";
+
+import {ClasschartsClient} from "./client"
+import { API_BASE_PARENT, BASE_URL } from "./consts";
+/**
+ * The base client
+ */
+export class ClasschartsClientParent extends ClasschartsClient {
```

Comment:
Small thing but it might be better if this is renamed to ClasschartsStudentClient, it just reads better.